DuckDuckGo/SettingsViewModel.swift

```diff
@@ -397,7 +402,8 @@ extension SettingsViewModel {
     private func setupNotificationObservers() {
         signOutObserver = NotificationCenter.default.addObserver(forName: .accountDidSignOut, object: nil, queue: .main) { [weak self] _ in
             if #available(iOS 15.0, *) {
-                self?.signOutUser()
+                guard let strongSelf = self else { return }
+                Task { await strongSelf.setupSubscriptionEnvironment() }
```

Comment:
Will it have any effect, the `setupSubscriptionEnvironment()` will return at first guard due to the lack of access token.